tests/codes/eurocode/nen_en_1992_1_1_c2_2011/chapter_4_durability_and_cover/test_table_4_3.py

```diff
@@ -0,0 +1,179 @@
+)
+
+
+class TestConcreteStructuralClassCalculator:
+    """Unit tests for the ConcreteStructuralClassCalculator class."""
+
+    @pytest.mark.parametrize(
+        ("design_working_life", "expected_structural_class", "expected_explanation"),
+        [(50, 4, "(50 years)"), (75, 5, "(75 years)"), (100, 6, "(100 years)")],
+    )
+    def test_structural_class_delta_design_working_life(
+        self, design_working_life: YEARS, expected_structural_class: int, expected_explanation: str
+    ) -> None:
+        """Test the _structural_class_delta_design_working_life method of the ConcreteStructuralClassCalculator class."""
+        calculator = ConcreteStructuralClassCalculator(
+            DUMMY_EXPOSURE_CLASSES,
+            design_working_life,
+            ConcreteMaterial(ConcreteStrengthClass("C20/25")),
+            False,
+            False,
+        )
+
+        calculator._structural_class_delta_design_working_life()  # noqa: SLF001
+        calculator._calculated = True  # noqa: SLF001
+
+        assert calculator.structural_class == expected_structural_class
+        assert calculator.explanation.endswith(expected_explanation)
+
+    @pytest.mark.parametrize(
+        ("plate_geometry", "expected_structural_class", "expected_explanation"),
+        [(True, 3, "(plate geometry)"), (False, 4, "(no plate geometry)")],
+    )
+    def test_structural_class_delta_plate_geometry(self, plate_geometry: bool, expected_structural_class: int, expected_explanation: str) -> None:
+        """Test the _structural_class_delta_plate_geometry method of the ConcreteStructuralClassCalculator class."""
+        calculator = ConcreteStructuralClassCalculator(
+            DUMMY_EXPOSURE_CLASSES,
+            50,
+            ConcreteMaterial(ConcreteStrengthClass("C20/25")),
+            plate_geometry,
+            False,
+        )
+
+        calculator._structural_class_delta_plate_geometry()  # noqa: SLF001
+        calculator._calculated = True  # noqa: SLF001
+
+        assert calculator.structural_class == expected_structural_class
+        assert calculator.explanation.endswith(expected_explanation)
+
+    @pytest.mark.parametrize(
+        ("quality_control", "expected_structural_class", "expected_explanation"),
+        [(True, 3, "(quality control)"), (False, 4, "(no quality control)")],
+    )
+    def test_structural_class_delta_quality_control(self, quality_control: bool, expected_structural_class: int, expected_explanation: str) -> None:
+        """Test the _structural_class_delta_quality_control method of the ConcreteStructuralClassCalculator class."""
+        calculator = ConcreteStructuralClassCalculator(
+            DUMMY_EXPOSURE_CLASSES,
+            50,
+            ConcreteMaterial(ConcreteStrengthClass("C20/25")),
+            False,
+            quality_control,
+        )
+
+        calculator._structural_class_delta_quality_control()  # noqa: SLF001
+        calculator._calculated = True  # noqa: SLF001
```

Comment:
This feels like a workaround. The calculator shouldn't be dependent of a private interface to get the final calculations.

Reply:
changed it.